hw/ip/pattgen/data/pattgen.hjson

```diff
@@ -0,0 +1,144 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+{ name: "pattgen"
+  clock_primary: "clk_fixed"
+  bus_device: "tlul"
+  bus_host: "none"
+  estimated_gate: "1k"
+  // OUTPUT pins
+  available_output_list: [
+    { name: "pda1_tx", desc: "Serial output data bit for pattern generation on channel 1" }
+    { name: "pcl1_tx", desc: "Bit-banging output clock for pattern generation on channel 1" }
+    { name: "pda2_tx", desc: "Serial output data bit for pattern generation on channel 2" }
+    { name: "pcl2_tx", desc: "Bit-banging output clock for pattern generation on channel 2" }
+  ]
+  // INTERRUPT pins
+  interrupt_list: [
+    { name: "patt_done1"
+      desc: "raise if pattern generation on channel 1 is complete"
+    }
+    { name: "patt_done2"
+      desc: "raise if pattern generation on channel 2 is complete"
+    }
+  ]
+
+  // REGISTER definition
+  regwidth: "32"
+  registers: [
+    // CTRL register
+    { name: "CTRL"
+      desc: "PATTGEN control register (Functions TBD)"
+      swaccess: "rw"
+      hwaccess: "hro"
+      fields: [
+        { bits: "0"
+          resval: "0"
+          name: "ENABLE"
+          desc: '''
+                Enable pattern generator functionality
+                '''
+        }
+      ]
+    }
+    { name: "START"
+      desc: "PATTGEN start pattern generation"
+      swaccess: "wo"
+      hwaccess: "hro"
+      hwqe: "true"
+      fields: [
+        { bits: "0"
+          resval: "0"
+          name: "START0"
+          desc: "Start generating pattern on channel 1"
+        }
+        { bits: "1"
+          resval: "0"
+          name: "START1"
+          desc: "Start generating pattern on channel 1"
+        }
+      ]
+    }
```

Comment:
just as a note for later, i think it would make sense for all the registers to make use of the `multireg` construct, that way if we decide to increase the number of groups in the future, it can easily be doe.

Reply:
@tjaychen agree to make use of `multireg` if we want to have more (>2) channels. Otherwise, I think, we can use isolate registers